frontend/src/pages/clusterSettings/ClusterSettings.tsx

```diff
@@ -0,0 +1,137 @@
+import * as React from 'react';
+import {
+  Button,
+  ButtonVariant,
+  Form,
+  FormGroup,
+  InputGroup,
+  InputGroupText,
+  InputGroupTextVariant,
+  PageSection,
+  PageSectionVariants,
+  Text,
+  TextInput,
+} from '@patternfly/react-core';
+import ApplicationsPage from '../ApplicationsPage';
+import { fetchClusterSettings, updateClusterSettings } from '../../services/clusterSettingsService';
+import { ClusterSettings } from '../../types';
+import './ClusterSettings.scss';
+
+const description = `Update global settings for all users.`;
+
+const DEFAULT_PVC_SIZE = 20;
+const DEFAULT_CONFIG: ClusterSettings = {
+  pvcSize: DEFAULT_PVC_SIZE,
+};
+
+const ClusterSettings: React.FC = () => {
+  const isEmpty = false;
+  const [loaded, setLoaded] = React.useState<boolean>(false);
+  const [loadError, setLoadError] = React.useState<Error>();
+  const [clusterSettings, setClusterSettings] = React.useState<ClusterSettings>(DEFAULT_CONFIG);
+  const MIN_PVC_SIZE = 1;
+  const MAX_PVC_SIZE = 16384;
+  const pvcDefaultBtnRef = React.useRef<HTMLButtonElement>();
+
+  React.useEffect(() => {
+    fetchClusterSettings()
+      .then((clusterSettings: ClusterSettings) => {
+        setLoaded(true);
+        setLoadError(undefined);
+        setClusterSettings(clusterSettings);
+      })
+      .catch((e) => {
+        setLoadError(e);
+      });
+  }, []);
+
+  const submitClusterSettings = async (newClusterSettings: ClusterSettings) => {
+    try {
+      const response: ClusterSettings = await updateClusterSettings(newClusterSettings);
```

Comment:
better to check `newClusterSettings` with `clusterSettings`, and only update it when it's changed, to reduce the times of network requests.